File: std/runtime/go/wj_json.go
// Windjammer std::json backend for Go (encoding/json).
// API mirrors windjammer_runtime::json (Rust).
package wjjson

import "encoding/json"

func Parse(s string) (map[string]interface{}, error) {
	var v map[string]interface{}
	if err := json.Unmarshal([]byte(s), &v); err != nil {
		return nil, err
	}
	return v, nil
}

func Stringify(v interface{}) (string, error) {
	b, err := json.Marshal(v)
	if err != nil {
		return "", err
	}
	return string(b), nil
}

File: std/runtime/js/wj_json.mjs
// Windjammer std::json backend for JavaScript (Node / browser).
export function parse(text) {
  return JSON.parse(text);
}

export function stringify(value) {
  return JSON.stringify(value);
}

export function isObject(value) {
  return value !== null && typeof value === "object" && !Array.isArray(value);
}

export function get(value, key) {
  if (value && typeof value === "object") return value[key];
  return undefined;
}
